CoAP causes build issues

In https://github.com/eclipse/thingweb.node-wot/pull/727 I wanted to update all version numbers from `0.8.0` to `0.8.1`.

Doing so I removed `package-lock.json` file and recreated it calling `npm install`.
Unfortunately this caused issues in binding-coap.

I wonder also why we use out-dated libraries.
e.g., "node-coap-client": "1.0.8" vs "node-coap-client": "2.0.2"

@JKRhb do you have a clue why this issue is happening?

![grafik](https://user-images.githubusercontent.com/11281461/161062324-0032aebf-d162-4e58-8e4a-cac42b35cbaf.png)

FYI, I see

```
  1) CoAP server implementation
       should read property:
     Uncaught TypeError: Cannot read properties of null (reading 'formIndex')
      at ExposedThing.handleUnobserveProperty (C:\Projects\WoT\thingweb\thingweb.node-wot-danielpeintner\packages\core\src\exposed-thing.ts:1277:25)
      at ObserveWriteStream.<anonymous> (src\coap-server.ts:369:43)
```

OR

```
  2) CoAP server implementation
       should read property:
     done() called multiple times in test <CoAP server implementation should read property> of file C:\Projects\WoT\thingweb\thingweb.node-wot-danielpeintner\packages\binding-coap\test\coap-server-test.ts; in addition, done() received error: AssertionError: expected '' to equal '"testValue"'
    at CoapServerTest.<anonymous> (C:\Projects\WoT\thingweb\thingweb.node-wot-danielpeintner\packages\binding-coap\test\coap-server-test.ts:68:82)
```

--- a/packages/binding-coap/package.json
+++ b/packages/binding-coap/package.json
@@ -15,9 +15,7 @@
     "types": "dist/coap.d.ts",
     "devDependencies": {
         "@testdeck/mocha": "^0.1.2",
-        "@types/bl": "^5.0.2",
         "@types/chai": "^4.2.18",
-        "@types/lru-cache": "^5.1.1",
         "@types/node": "16.4.13",
         "@typescript-eslint/eslint-plugin": "^4.30.0",
         "@typescript-eslint/parser": "^4.30.0",
@@ -39,7 +37,7 @@
         "@node-wot/core": "0.8.1",
         "@node-wot/td-tools": "0.8.1",
         "@types/node": "16.4.13",
-        "coap": "^1.0.3",
+        "coap": "^1.0.8",
         "node-coap-client": "1.0.8",
         "rxjs": "5.5.11",
         "slugify": "^1.4.5",

